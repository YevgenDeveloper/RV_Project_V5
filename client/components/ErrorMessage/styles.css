@import 'client/styles/variables';

.message {
  display: flex;
  position: relative;
  width: 100%;
  margin: auto;

  .inner {
    margin: auto;
    padding: 20px;
    text-align: center;
  }

  h1 {
    font-weight: bold;
    font-size: $font-size-large;
    color: $color-text-default;
    margin: 1em 0;
    user-select: text;
  }

  p {
    font-size: $font-size-default;
    color: $color-red;
    margin: 2em 0;
    user-select: text;
  }

  .button {
    display: block;
    margin: 1em auto;
    font-size: $font-size-default;
  }
}
