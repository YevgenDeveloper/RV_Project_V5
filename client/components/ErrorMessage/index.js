// @flow
import React from 'react';
import classNames from 'classnames/bind';
import { Button } from 'client/components';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  header: string,
  text?: string,
  className?: string,
  actionLabel?: string,
  action?: void => void,
};

const Message = (props: Props) => (
  <div className={cx(styles.message, props.className)}>
    <div className={styles.inner}>
      <h1>{props.header}</h1>
      <p>{props.text}</p>
      {props.action &&
        props.actionLabel && (
          <Button
            label={props.actionLabel}
            className={styles.button}
            onClick={props.action}
          />
        )}
    </div>
  </div>
);

Message.defaultProps = {
  text: undefined,
  className: undefined,
  actionLabel: undefined,
  action: undefined,
};

export default Message;
