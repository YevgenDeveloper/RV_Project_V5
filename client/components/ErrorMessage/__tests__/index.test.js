// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<ErrorMessage />);

  expect(wrapper).toMatchSnapshot();
});
