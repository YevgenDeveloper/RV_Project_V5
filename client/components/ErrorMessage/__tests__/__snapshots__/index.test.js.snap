// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner"
      >
        <h1 />
        <p />
      </div>,
      "className": "message",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <p />,
          undefined,
        ],
        "className": "inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        undefined,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner"
        >
          <h1 />
          <p />
        </div>,
        "className": "message",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <p />,
            undefined,
          ],
          "className": "inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
